Improve the guideline of Preparing gpg key (#356)
apache/spark-website: release-process.md

We need to generate a lot of random bytes. It is a good idea to perform
some other action (type on the keyboard, move the mouse, utilize the
disks) during the prime generation; this gives the random number
generator a better chance to gain enough entropy.
gpg: key 04B3B5C426A27D33 marked as ultimately trusted
gpg: revocation certificate stored as '/home/ubuntu/.gnupg/openpgp-revocs.d/08071B1E23C8A7E2CA1E891A04B3B5C426A27D33.rev'
public and secret key created and signed.

pub   rsa4096 2021-08-19 [SC]
      08071B1E23C8A7E2CA1E891A04B3B5C426A27D33
uid                      Jack (test) <Jack@mail.com>
sub   rsa4096 2021-08-19 [E]
```

Note that the last 8 digits (26A27D33) of the public key is the <a href="https://infra.apache.org/release-signing.html#key-id">key ID</a>.

<h4>Upload Key</h4>

After generating the public key, we should upload it to <a href="https://infra.apache.org/release-signing.html#keyserver">public key server</a>:

```
$ gpg --keyserver keys.openpgp.org --send-key 26A27D33
```

Please refer to <a href="https://infra.apache.org/release-signing.html#keyserver-upload">keyserver-upload</a> for details.

<h4>Update KEYS file with your code signing key</h4>

> The code signing key is exactly the same with the ASCII-armored public key mentioned above.

Reviewer: The `ASCII-armored public key` seems not being mentioned anymore

Author: Oh. that was mentioned in the removed second key loading way. Let me move it here.

```
:::console
$ gpg --full-gen-key
gpg (GnuPG) 2.0.12; Copyright (C) 2009 Free Software Foundation, Inc.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Please select what kind of key you want:
   (1) RSA and RSA (default)
   (2) DSA and Elgamal
   (3) DSA (sign only)
   (4) RSA (sign only)
Your selection? 1
RSA keys may be between 1024 and 4096 bits long.
What keysize do you want? (2048) 4096
Requested keysize is 4096 bits
Please specify how long the key should be valid.
         0 = key does not expire
      <n>  = key expires in n days
      <n>w = key expires in n weeks
      <n>m = key expires in n months
      <n>y = key expires in n years
Key is valid for? (0) 
Key does not expire at all
Is this correct? (y/N) y

GnuPG needs to construct a user ID to identify your key.

Real name: Robert Burrell Donkin
Email address: rdonkin@apache.org
Comment: CODE SIGNING KEY
You selected this USER-ID:
    "Robert Burrell Donkin (CODE SIGNING KEY) <rdonkin@apache.org>"

Change (N)ame, (C)omment, (E)mail or (O)kay/(Q)uit? O
You need a Passphrase to protect your secret key.
```

> <h4>Upload Key</h4>

Reviewer: This new section seems duplicated with https://infra.apache.org/openpgp.html

Author: These are necessary steps that are abstracted from https://infra.apache.org/openpgp.html, which includes massive and too much irrelevant information.

After generating the key, we should upload the public key to a <a href="https://infra.apache.org/release-signing.html#keyserver">public key server</a>.
Upload the public key either by:

(<b>Recommended</b>)
First, export all public keys to ASCII-armored public key by
```
:::console
$ gpg --export --armor 
```
or export the specific public key if you know the <a href="https://infra.apache.org/release-signing.html#key-id">key ID</a>, e.g.,
```
:::console

> $ gpg --export --armor AD741727

Reviewer 2: is there a way to find the id of the newly created CODE SIGNING KEY?

Author: Added an example of how to find the key id.

```
(Please refer to <a href="https://infra.apache.org/openpgp.html#export-public-key">export-public-key</a> for details.)

Second, copy-paste your ASCII-armored public key to <a href="http://keyserver.ubuntu.com:11371/#submitKey">OpenPGP Keyserver</a> and submit.

or

Use gpg command to upload, e.g.,

```

> $ gpg --send-key B13131DE2

Reviewer 2: Can we use the same id in the previous example `AD741727`?

some other action (type on the keyboard, move the mouse, utilize the
disks) during the prime generation; this gives the random number
generator a better chance to gain enough entropy.
We need to generate a lot of random bytes. It is a good idea to perform
some other action (type on the keyboard, move the mouse, utilize the
disks) during the prime generation; this gives the random number
generator a better chance to gain enough entropy.
gpg: key 04B3B5C426A27D33 marked as ultimately trusted
gpg: revocation certificate stored as '/home/ubuntu/.gnupg/openpgp-revocs.d/08071B1E23C8A7E2CA1E891A04B3B5C426A27D33.rev'
public and secret key created and signed.

pub   rsa4096 2021-08-19 [SC]
      08071B1E23C8A7E2CA1E891A04B3B5C426A27D33
uid                      Jack (test) <Jack@mail.com>
sub   rsa4096 2021-08-19 [E]
```

Note that the last 8 digits (26A27D33) of the public key is the <a href="https://infra.apache.org/release-signing.html#key-id">key ID</a>.

<h4>Upload Key</h4>

After generating the public key, we should upload it to a <a href="https://infra.apache.org/release-signing.html#keyserver">public key server</a>.
You can upload:

either use gpg command:

```

> $ gpg --send-key 26A27D33 --keyserver keys.openpgp.org

Reviewer: nit: change the order of param so that it is easier to modify the key name 
```
gpg --keyserver keys.openpgp.org --send-key 26A27D33
```

```

or copy-paste the ASCII-armored public key to <a href="http://keyserver.ubuntu.com:11371/#submitKey">OpenPGP Keyserver</a>.
The ASCII-armored public key can be generated by:

```
:::console
$ gpg --export --armor 26A27D33
```

Please refer to <a href="https://infra.apache.org/release-signing.html#keyserver-upload">keyserver-upload</a> for details.

<h4>Update KEYS file with your code signing key</h4>

The code signing key is exactly the same with the ASCII-armored public key mentioned above.

> You should append it to <a href="https://dist.apache.org/repos/dist/dev/spark/KEYS">KEYS</a> by:

Reviewer 2: ```suggestion
You should append it to the KEYS file by:
```

It doesn't seem necessary to add url for `KEYS`. People need to run the svn command below anyway.

Please specify how long the key should be valid.
         0 = key does not expire
      <n>  = key expires in n days
      <n>w = key expires in n weeks
      <n>m = key expires in n months
      <n>y = key expires in n years
Key is valid for? (0) 
Key does not expire at all
Is this correct? (y/N) y

GnuPG needs to construct a user ID to identify your key.

Real name: Robert Burrell Donkin
Email address: rdonkin@apache.org
Comment: CODE SIGNING KEY
You selected this USER-ID:
    "Robert Burrell Donkin (CODE SIGNING KEY) <rdonkin@apache.org>"

Change (N)ame, (C)omment, (E)mail or (O)kay/(Q)uit? O
You need a Passphrase to protect your secret key.
```

<h4>Upload Key</h4>

After generating the key, we should upload the public key to a <a href="https://infra.apache.org/release-signing.html#keyserver">public key server</a>.
Upload the public key either by:

(<b>Recommended</b>)
First, export all public keys to ASCII-armored public key by
```
:::console
$ gpg --export --armor 
```
or export the specific public key if you know the <a href="https://infra.apache.org/release-signing.html#key-id">key ID</a>, e.g.,
```
:::console
$ gpg --export --armor AD741727
```

> Here's an example of how to get the key ID for your public key:

Reviewer 2: Can we move it to the `Generate Key` section? I think the release manager should get the key id right after he/she generates the gpg key.
